### DISCUSSION TITLE
Can't get dependencies to work in the extension improvement branch

### DISCUSSION THREAD

**Discussion by VendoAU** [Minestom/Minestom#873] [Help] [answered]
Hi, I've made two completely blank extensions to test this and I still cannot get it to work.
When I start my server it says: `Unknown extension: Test (dependency of Test2)`
I am using commit b63a73516d

Here is the extension.json for both extensions
Test:
```json
{
  "entrypoint": "com.vendoau.test.Test",
  "name": "Test",
  "version": "1.0.0"
}
```
Test2:
```json
{
  "entrypoint": "com.vendoau.test2.Test2",
  "name": "Test2",
  "version": "1.0.0",
  "dependencies": [
    "Test"
  ]
}
```

Am I doing something wrong or is this just broken?
Thanks.

**Comment by mworzala** [comment]
It doesn't look like anything is wrong - and should be working. I think I can look into this in a few hours. In the mean time, is there anything else in the logs (for example if Test failed to load this could potentially happen)? How are the extensions being loaded (built jars in the extension dirs, "autoscan", indev properties)?

**Reply by VendoAU** [reply]
I have the built extension jars in the extensions folder of the server and I am running it in IntelliJ

This is the only thing in my main server class:
```java
final MinecraftServer minecraftServer = MinecraftServer.init();
minecraftServer.start("0.0.0.0", 25565);
```

This is the entire log:
```
7:08:15 pm: Executing task ':Server.main()'...

> Task :compileJava UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE

> Task :Server.main()
[main] [19:08:16] (ServerProcessImpl.start) - INFO - Starting Minestom server.
Exception in thread "main" java.lang.IllegalStateException: Unknown extension: Test (dependency of Test2)
	at net.minestom.server.extensions.ExtensionManager.computeLoadOrder(ExtensionManager.java:175)
	at net.minestom.server.extensions.ExtensionManager.start(ExtensionManager.java:139)
	at net.minestom.server.ServerProcessImpl.start(ServerProcessImpl.java:218)
	at net.minestom.server.MinecraftServer.start(MinecraftServer.java:363)
	at net.minestom.server.MinecraftServer.start(MinecraftServer.java:368)
	at com.vendoau.testserver.Server.main(Server.java:62)
```

**Comment by BitCrack** [comment, marked as answer]
I believe this is related to https://github.com/Minestom/Minestom/pull/855

**Reply by VendoAU** [reply]
Using the PR solves the issue.
Thanks for your help.
